### DISCUSSION TITLE
White screen at enter in login

### DISCUSSION THREAD

**Discussion by cleanet** [coder/code-server#4030] [Q&A]
Hi!
I have a problem with code-server.....
I installed code-server in a server Debian in Raspberry Pi 7b+ [4.19.0-17-arm64 #1 SMP Debian 4.19.194-3 (2021-07-18) aarch64 GNU/Linux]. In the install never problem, the service code-server@USER is enabled y start successfully....

This IDE, I want deploy it, in server apache2 by port 10192, so installed apache2, and I created a VirtualHost, his configuration is follow:
```
<VirtualHost *:10192>
        ServerName vscode
        ServerAdmin webmaster@localhost
        DocumentRoot /home/vscode
        RewriteEngine On
        RewriteCond %{HTTP:UPGRADE} ^WebSocket$ [NC,OR]
        RewriteCond %{HTTP:CONNECTION} ^Upgrade$ [NC]
        RewriteRule .* ws://127.0.0.1:8080%{REQUEST_URI} [P,QSA,L]
        <Location />
                Require all granted
                ProxyPass http://127.0.0.1:8080/
                ProxyPassReverse http://127.0.0.1:8080/
                ProxyPassReverseCookieDomain 127.0.0.1 www.losheredia.local
        </Location>
</VirtualHost>
```
I run good, I go to http://127.0.0.1:10192/ and it show the code-server's login.
![image](https://user-images.githubusercontent.com/59420450/130686231-64f35652-a6bd-4aa2-afdd-a4ce9d42e114.png)
 
But... at log in....the screen is white.....

The file `~/.config/code-server/config.yaml` his configurations are followed:
```
bind-addr: 127.0.0.1:8080
auth: password
password: b29599a434a2390d05eb1061
cert: false
```
and the command that the service `code-server@USER.service` uses for start code-server is follow: `/usr/lib/code-server/lib/node /usr/lib/code-server --bind-addr 127.0.0.1:8080 --user-data-dir /var/lib/code-server --auth password`.

Although. Is the service normal that run two times the same command?
```
● code-server@USER.service - code-server
   Loaded: loaded (/lib/systemd/system/code-server@.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2021-08-24 14:36:29 UTC; 6h ago
 Main PID: 425 (node)
    Tasks: 44 (limit: 989)
   Memory: 230.7M
   CGroup: /system.slice/system-code\x2dserver.slice/code-server@USER.service
           ├─ 425 /usr/lib/code-server/lib/node /usr/lib/code-server --bind-addr 127.0.0.1:8080 --user-data-dir /var/lib/code-server --auth password
           ├─1089 /usr/lib/code-server/lib/node /usr/lib/code-server --bind-addr 127.0.0.1:8080 --user-data-dir /var/lib/code-server --auth password
           ├─2941 /usr/lib/code-server/lib/node /usr/lib/code-server/lib/vscode/out/vs/server/fork
           └─2956 /usr/lib/code-server/lib/node /usr/lib/code-server/lib/vscode/out/bootstrap-fork --type=ptyHost

Aug 24 14:36:29 servheredia systemd[1]: Started code-server.
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.708Z] info  code-server 3.11.1 c680aae973d83583e4a73dc0c422f44021f0140e
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.718Z] info  Using user-data-dir /var/lib/code-server
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.837Z] info  Using config file ~/.config/code-server/config.yaml
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.838Z] info  HTTP server listening on http://127.0.0.1:8080
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.839Z] info    - Authentication is enabled
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.840Z] info      - Using password from $PASSWORD
Aug 24 14:36:51 servheredia code-server[425]: [2021-08-24T14:36:51.840Z] info    - Not serving HTTPS
Aug 24 19:07:25 servheredia code-server[425]:  WARN No ptyHost heartbeat after 6 seconds
```

In the navegator, when I was logged, and the screen is white...I looked the inspect Element, and in the console there are very errors (I imagine some are disenchanted of others).
![Screenshot from 2021-08-24 22-48-04](https://user-images.githubusercontent.com/59420450/130687806-e4dc9368-fd8c-4fa8-9eed-20f801cbf4dd.png)

And in the network...
![Screenshot from 2021-08-24 22-50-54](https://user-images.githubusercontent.com/59420450/130688138-b64010d2-5bf7-49a3-95a5-fcbdfc97f266.png)
![Screenshot from 2021-08-24 22-50-43](https://user-images.githubusercontent.com/59420450/130688149-13d7666f-c711-4f4b-ae55-854c67fa5352.png)

Why are there many wrongs? And is there some solution?

Thank you very much.

**Comment by jsjoeio** [comment]
Hmm...it seems like something is going wrong with the WebSocket connection. I am not familiar with this type of setup but maybe @code-asher is

**Reply by cleanet** [reply]
Thank you very much, I will ask her it

**Reply by jsjoeio** [reply]
*him !

**Reply by cleanet** [reply]
Thanks very much by the correction! I am not very good in English, and the grammar more or less, I do it I can, I appreciate any thing.

Is "him" and not "her", Because "him" is in male while that "her" is female.... Thank you.

**Reply by jsjoeio** [reply]
> Thanks very much by the correction! I am not very good in English, and the grammar more or less, I do it I can, I appreciate any thing.

No worries at all! English is very hard IMO and has weird rules. My word of advice: use their/them when you don't know someone's preferred pronouns

**Reply by cleanet** [reply]
Yes, the English is very difficult, but while the communication is readable and understandable.... Thank you by the tip.

**Comment by code-asher** [comment]
Hmm often when the web socket fails to connect it's because VS Code has crashed, and that's usually because of a problem with native modules. The `WARN No ptyHost heartbeat after 6 seconds` supports this theory.

The dual processes are expected since code-server calls itself (this is so it can restart itself).

Running code-server with the `-vvv` flag might reveal more information (although with these crashes it usually doesn't).

How was code-server installed? Through the install script?

**Reply by cleanet** [reply]
I edited the service file and I added the `-vvv` flag, now print the wrongs more detailed, in the log `/var/log/syslog` print the follow:
![Screenshot from 2021-08-25 23-15-45](https://user-images.githubusercontent.com/59420450/130866015-44468f42-b39f-4934-8317-bc7574e769ad.png)

And in the browser console follow it:
![Screenshot from 2021-08-25 23-21-31](https://user-images.githubusercontent.com/59420450/130866468-44d11262-0d6a-495e-90e8-95609a47a620.png)
![Screenshot from 2021-08-25 23-21-52](https://user-images.githubusercontent.com/59420450/130866467-385e9a97-1576-49a1-9499-87c642cce229.png)
In the console tells that It can't read the files :`/var/lib/code-server/User/settings.json`,`/var/lib/code-server/User/tasks.json`, but, the files doesn't exist! How? I don't understand it, missing very files....

The code-server script installer `install.sh` created me the folder User, but this there are only thow folter empties.

I installed the code-server with the command: `curl -fsSL https://code-server.dev/install.sh | sh`

In the path `/var/lib/code-server/` the permissions **recursive** are: root:ww-data rwxrwx---
![Screenshot from 2021-08-25 23-42-52](https://user-images.githubusercontent.com/59420450/130868602-415b1d25-943f-4a85-b60f-4a12120f57ac.png)

Here the code-server folders tree:
```
code-server/
|-- User
|   |-- globalStorage
|   `-- workspaceStorage
|-- extensions
|-- logs
|   |-- 20210823T180020
|   |   |-- remoteagent.log
|   |   `-- telemetry.log
|   |-- 20210824T190718
|   |   |-- remoteagent.log
|   |   `-- telemetry.log
|   |-- 20210825T204453
|   |   |-- remoteagent.log
|   |   `-- telemetry.log
|   `-- 20210825T205211
|       |-- remoteagent.log
|       `-- telemetry.log
`-- machineid

```
And I edited the `/etc/apache2/sites-avaible/vscode.conf` and changed the DocumentRoot to `/var/lib/code-server`.
```
        ServerName vscode
        ServerAdmin webmaster@localhost
        DocumentRoot /var/lib/code-server

        # Available loglevels: trace8, ..., trace1, debug, info, notice, warn,
        # error, crit, alert, emerg.
        # It is also possible to configure the loglevel for particular
        # modules, e.g.
        #LogLevel info ssl:warn
        <Directory /var/lib/code-server>
                AllowOverride all
                Require all granted
        </Directory>
```

And I did all this, and `www.losheredia.local:10192` is in white....

**Reply by code-asher** [reply]
Thank you for posting all this!

You can ignore the `settings.json` errors, these look like they are being caused by the websocket failure.

Right now my best guess is that the arm64 package we provide does not work on the Raspberry Pi 7b+.

One way to test this theory would be to install code-server with yarn `yarn global add code-server` but this will require build dependencies and may take a long time! Unfortunately I do not have a Pi to test our package on.

**Reply by cleanet** [reply]
![Screenshot from 2021-08-28 22-09-42](https://user-images.githubusercontent.com/59420450/131229838-cec4c7a3-ac6b-4caf-8b5f-b9c3b7c1b243.png)

Hi! Thank you by you aswer. You don't worry, the command `yarn` with argument `global` doesn't work me, throw a fail. I repeat, you don't worry. I only would want a server IDE, but the code-server throw me very wrongs, so, if don't exists other IDE server for install locally...., else doesn't happen nothing.
Thank you very much.

**Reply by code-asher** [reply]
Oh strange I wonder if you have an old version of `yarn`? You could try npm instead: `npm install -g code-server`.

If you run this please make sure to keep the `npm` logs in case something goes wrong so we can look at them.

I think we need more logging or something to help diagnose these silent failures.

**Reply by jsjoeio** [reply]
Yes, before you install with `npm`, please read [this](https://coder.com/docs/code-server/v3.11.1/npm#debugging-install-issues-with-npm). It will tell you how to enable logging, which we'll need to help you troubleshoot further
